### PR TITLE
Implemented the Array.prototype.some method.

### PR DESCRIPTION
**Pull request by HalidOdat**
I implemented the `Array.prototype.some` based on the [spec](https://tc39.es/ecma262/#sec-array.prototype.some). I used `Array.prototype.every` method as a reference point, since they are so similar.

### CODE REVIEW

File: boa/src/builtins/array/mod.rs [hunk 1]
```diff
@@ -778,6 +778,43 @@ pub fn filter(this: &Value, args: &[Value], interpreter: &mut Interpreter) -> Re
...
+    let max_len: i32 = from_value(this.get_field_slice("length")).unwrap();
+    let mut len = max_len;
+    while i < len {
+        let element = this.get_field_slice(&i.to_string());
+        let arguments = vec![element.clone(), to_value(i), this.clone()];
+        let result = interpreter.call(callback, &this_arg, arguments)?.is_true();
+        if result {
+            return Ok(to_value(true));
+        }
+        len = min(max_len, from_value(this.get_field_slice("length")).unwrap());
```
Comment (Razican) [inline boa/src/builtins/array/mod.rs:812]: Why do we update the length here?

Comment (HalidOdat) [reply boa/src/builtins/array/mod.rs:812]: Its because the callback can mutate the array and we need to update it.

**Comment by HalidOdat** [comment]
I think this branch is ready to merge. :)

**Comment by jasonwilliams** [comment]
At some point https://tc39.es/ecma262/#sec-iscallable will need to be implemented to check the passed in arg is a function, but i think that can be done separately considering there's an update to how functions will work

**Comment by HalidOdat** [comment]
Added comment describing the updating of the length.

**Pull request merged**